qiita_db/util.py

```diff
@@ -106,6 +107,8 @@ def typecast_string(string):
     The function first tries to convert to an int. If that fails, it tries to
     convert to a float. If that fails it returns the original string.
     """
+    if isinstance(string, datetime):
```

Comment:
could add an explicit test and raise, or leave it up to the caller to determine what to do. the current method returns the data silently without casting if something goes wrong, which is arguably worse